### ISSUE TITLE
CTC forced alignment: word-level timestamps for SenseVoice & Wav2Vec2

### ISSUE THREAD

**Issue by SRjoeee** [findit-studio/mlxrs#385]
## Overview

Add CTC forced alignment (Viterbi) as a generic post-processing capability on the `CtcModel` trait, enabling word-level timestamps for **SenseVoice** and **Wav2Vec2** (and any future CTC model).

Whisper already has timestamps via cross-attention DTW (#339). This closes the gap for the remaining STT models.

## Background: Three timestamp mechanisms

| Mechanism | Models | How it works | mlxrs support |
|-----------|--------|-------------|---------------|
| Time tokens in vocabulary | Whisper | Model outputs `<\|0.00\|>` tokens natively | ✅ DTW (#339) |
| CIF predictor | Paraformer | Built-in trained predictor fires at token boundaries | N/A (not ported) |
| **CTC forced alignment** | **SenseVoice, Wav2Vec2** | Viterbi DP on CTC log-probs + decoded token IDs | ❌ **This issue** |

CTC forced alignment is a **pure algorithm** — no model changes, no training, no new weights. It takes the CTC log-probs that `CtcModel::logits()` already produces and the token IDs from the greedy decode, and finds the optimal frame-level alignment via Viterbi dynamic programming.

## Reference implementation

FunASR's `ctc_forced_align` (`funasr/models/sense_voice/utils/ctc_alignment.py`):

```python
def ctc_forced_align(log_probs, targets, input_lengths, target_lengths, blank=0, ignore_id=-1):
    # Viterbi forced alignment over CTC log-probs
    # Input:  log_probs (B, T, C), targets (B, L)
    # Output: alignment (B, T) — per-frame token ID assignment
```

FunASR's SenseVoice timestamp pipeline (`model.py:1036-1074`):

```python
# 1. Re-tokenize decoded text → token IDs
# 2. Run ctc_forced_align(log_probs, token_ids)
# 3. Convert frame indices → timestamps:
#    ts_left = max((_start * 60 - 30) / 1000, 0)   # 60 = lfr_n * frame_shift_ms
#    ts_right = min((_end * 60 - 30) / 1000, ts_max * 60 - 30) / 1000)
# 4. Merge sub-word tokens → word-level timestamps via post()
```

The timestamp formula uses:
- `60` = LFR stride (6) × fbank frame shift (10ms) = 60ms per LFR frame
- `30` = centering offset (60/2)
- These are SenseVoice-specific; Wav2Vec2 uses different frame rates

### 1. Add `ctc_forced_align` to the STT generate module

```rust
// audio/stt/generate.rs

/// Viterbi forced alignment: find the optimal frame-to-token mapping
/// given CTC log-probs and a known token sequence.
///
/// Input: `log_probs` (T', vocab) — CTC log-probs from `CtcModel::logits()`
///        `token_ids` — the decoded token sequence (no blanks)
/// Output: `alignment` (T') — per-frame token index (0 = blank)
pub fn ctc_forced_align(log_probs: &Array, token_ids: &[u32]) -> Result<Vec<u32>> {
    // Viterbi DP over the CTC trellis
    // O(T' * len(token_ids)) — fast, no model calls
}
```

### 2. Add `timestamped_transcribe` to `CtcModel` trait

```rust
// audio/stt/model.rs

/// A word-level timestamp: the token text plus its [start, end] in seconds.
pub struct WordTimestamp {
    pub word: String,
    pub start_sec: f32,
    pub end_sec: f32,
}

/// A transcription with word-level timestamps.
pub struct TimestampedTranscription {
    pub text: String,
    pub words: Vec<WordTimestamp>,
    pub language: Option<String>,
}

/// Extension method on CtcModel — default implementation using forced alignment.
fn timestamped_transcribe(
    &self,
    waveform: &Array,
    frame_shift_ms: f32,   // e.g. 60.0 for SenseVoice, 20.0 for Wav2Vec2
    frame_offset_ms: f32,  // e.g. 30.0 for SenseVoice centering
) -> Result<TimestampedTranscription> {
    let logits = self.logits(waveform)?;           // (T', vocab)
    let ids = greedy_collapse(&logits, self.blank_id())?;
    let alignment = ctc_forced_align(&logits, &ids)?;
    let words = merge_alignment_to_words(&alignment, &ids, self, frame_shift_ms, frame_offset_ms)?;
    Ok(TimestampedTranscription { text: ..., words, ... })
}
```

### 3. Add sub-word → word merge post-processing

```rust
/// Merge sub-word tokens into word-level timestamps.
/// SentencePiece `▁` marks word boundaries; consecutive ASCII alpha tokens merge.
fn merge_alignment_to_words(
    alignment: &[u32],
    token_ids: &[u32],
    model: &dyn CtcModel,
    frame_shift_ms: f32,
    frame_offset_ms: f32,
) -> Result<Vec<WordTimestamp>> {
    // Walk the alignment, group consecutive frames with the same token,
    // merge sub-words by the ▁ prefix rule (matching FunASR's post())
}
```

### 4. Per-model frame rate configuration

| Model | frame_shift_ms | frame_offset_ms | Notes |
|-------|---------------|----------------|-------|
| SenseVoice | 60.0 | 30.0 | `lfr_n(6) * frame_shift(10ms)` = 60ms |
| Wav2Vec2 | 20.0 | 10.0 | fbank frame_shift=20ms, no LFR |

Each model provides these via its config or a trait method.

## Files to modify

- `mlxrs/src/audio/stt/model.rs` — add `WordTimestamp`, `TimestampedTranscription`, `timestamped_transcribe` default method
- `mlxrs/src/audio/stt/generate.rs` — add `ctc_forced_align`, `merge_alignment_to_words`
- `mlxrs/src/audio/stt/models/sensevoice/model.rs` — implement `timestamped_transcribe` with SenseVoice frame rate
- `mlxrs/src/audio/stt/models/wav2vec2.rs` — implement `timestamped_transcribe` with Wav2Vec2 frame rate
- Tests: oracle tests with known alignment inputs

## Scope

- **In scope**: CTC Viterbi forced alignment, sub-word merge, word-level timestamps for SenseVoice and Wav2Vec2
- **Out of scope**: Paraformer CIF predictor (requires full model port, no mlx-audio upstream reference), sentence-level punctuation timestamps

## Test plan

1. Unit test `ctc_forced_align` with a synthetic (T'=10, vocab=4) log-prob grid and known token sequence — verify alignment matches hand-computed Viterbi result
2. Unit test `merge_alignment_to_words` with a known alignment + token list — verify word boundaries and timestamps
3. Integration test on the real `mlx-community/SenseVoiceSmall` checkpoint with a short audio — verify timestamps are non-overlapping, monotonically increasing, and cover the full utterance duration
4. Cross-validate against FunASR output on the same audio — timestamps should be close (within ±1 frame = ±60ms)
